sawtooth-sdk-transaction-processor/src/main/java/sawtooth/sdk/processor/TransactionProcessor.java

```diff
@@ -14,81 +14,81 @@
 
 package sawtooth.sdk.processor;
 
-import com.google.protobuf.ByteString;
+import java.util.ArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
 import com.google.protobuf.InvalidProtocolBufferException;
 
 import sawtooth.sdk.messaging.Future;
 import sawtooth.sdk.messaging.Stream;
 import sawtooth.sdk.messaging.ZmqStream;
-import sawtooth.sdk.processor.exceptions.InternalError;
-import sawtooth.sdk.processor.exceptions.InvalidTransactionException;
 import sawtooth.sdk.processor.exceptions.ValidatorConnectionError;
 import sawtooth.sdk.protobuf.Message;
 import sawtooth.sdk.protobuf.PingResponse;
 import sawtooth.sdk.protobuf.TpProcessRequest;
-import sawtooth.sdk.protobuf.TpProcessResponse;
 import sawtooth.sdk.protobuf.TpRegisterRequest;
 import sawtooth.sdk.protobuf.TpUnregisterRequest;
 import sawtooth.sdk.protobuf.TransactionHeader;
 
-import java.util.ArrayList;
-import java.util.concurrent.TimeoutException;
-import java.util.logging.Level;
-import java.util.logging.Logger;
-
 /** Sawtooth transaction processor. */
 public class TransactionProcessor implements Runnable {
 
   /** Logging class for this processor. */
   private static final Logger LOGGER = Logger.getLogger(TransactionProcessor.class.getName());
 
+  /**
+   * Default time to wait for the threads to shutdown.
+   */
+  public static final long DEAULT_SHUTDOWN_TIMEOUT = 30;
+
   /** Streaming class for this processor. */
   private Stream stream;
 
   /** List of transaction handlers for this processor. */
   private ArrayList<TransactionHandler> handlers;
 
-  /** The current message for this processor. */
-  private Message currentMessage;
-
   /** Whether or not this processor has been registered. */
   private boolean registered;
 
+  /** Flag signifying whether or not this thread should keep running. */
+  private boolean keepRunningFlag;
```

Comment:
I think this should be called `running` or `stopped` (with the appropriate boolean for either name) and the corresponding function as `isRunning` or `isStopped`.

Reply:
This gets refactored later into an AtomicBoolean,

Reply:
renamed